apps/jobbrowser/src/jobbrowser/templates/job_browser.mako

```diff
@@ -3026,6 +3026,25 @@ ${ commonheader("Job Browser", "jobbrowser", user, request) | n,unicode }
         });
       };
 
+      self._rewriteKnoxUrls = function (data) {
+        if (data.app.type === 'SPARK') {
+          data.app.properties.metadata.forEach(function (item) {
```

Comment:
Are properties and metadata always defined?

Reply:
@JohanAhlen I have now added quite a few null checks? Is there a cleaner way to do this?

Reply:
I know the feeling... this would be the way to go here I'm afraid.

Something like data?.app?.properties?.metadata would be cleaner but sadly for inline js in .mako files we can't use the more modern ways as it doesn't go through webpack/babel (only js/ts under desktop/core/src/desktop/js do).